### PR TITLE
cmdline utility, pip+conda packaging, API simplifications

### PR DESCRIPTION
**Pull request by mjuric**
This PR brings a number of changes designed to make installing and using the ADAM client easier. The details of the changes are in individual commit messages, I'll only extract the highlights here:

* ADAM client can now be installed with conda:
  ```
   conda install -c asteroid-institute adam
  ```
   try it!
* We now have a new command-line utility, `adamctl`, that implements the functionality of the old `demo_config.ipynb` notebook. Specifically, one can log in by running:
  ```bash
  adamctl login
  adamctl config envs.prod.workspace "uuid-received-from-@AstrogatorJohn"
  ```
  This largely replaces the current onboarding guide.
* We store configs into `~/.config/adam/config` (per [XDG base dirs](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) guidelines), and in YAML format (for readability). As this is a well-known dir, there are no more need for path hacking boilerplate at the beginning of demo notebooks.
* I reworked the API of some of the configuration classes to make it more streamlined. This affects all notebooks that use these classes (I updated `delta_v_demo.ipynb` and `single_run_demo.ipynb`).
* ADAM client is now wrapped in standard `setup.py` and should be developed by running `python setup.py develop` (rather than manually hacking the `$PYTHONPATH`) 
* Version numbers are auto-generated from git tags.
* `README.md` has now been updated.

This turned into a rather massive PR w. a large number of wide-reaching client changes, but I hope they make sense (and I'll be happy to tweak it as necessary if there are concerns about any part of it). The goal was to make the client more Pythonic and improve the foundations for future development (e.g., group/permission management could next be added to `adamctl`, etc).

### CODE REVIEW

File: adam/tests/config_manager_test.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 from adam import ConfigManager
-import json
+import yaml
```
Comment (carise) [inline adam/tests/config_manager_test.py]: super nit: could you alphasort the imports? Something like what PEP-8 suggests -- https://www.python.org/dev/peps/pep-0008/#imports

File: adam/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import pytest
+
```
Comment (carise) [inline adam/tests/conftest.py]: Where is this file used?

File: adam/tests/config_manager_test.py (hunk 1 above)
Comment (mjuric) [reply adam/tests/config_manager_test.py]: Fixed! N.b. [flake8-import-order](https://github.com/PyCQA/flake8-import-order) found another 142 issues in existing code that I'll leave for later.

File: adam/tests/conftest.py (hunk 2 above)
Comment (mjuric) [reply adam/tests/conftest.py]: pytest framework imports it automagically before running any of the tests in `adam/tests` (and any subdrectories). It's a way to define fixtures and [generally hook into pytests internal processing](https://stackoverflow.com/a/34520971).

**Comment by mjuric** [comment]
@AstrogatorJohn Can I get the credentials needed to update the encrypted Adam config file Travis uses for the tests?

**Comment by carise** [comment]
resolves: #88 
resolves: #89

**Pull request merged**